### PR TITLE
Fix: Fixed create file button when entering text with Chinese input flyout

### LINKED ISSUE #9563: Bug: Create file button is disabled when using Chinese input flyout
### Description

When I try to create a folder or file and try to type the name in Chinese, I can't create it immediately(the button below remains grey.).
Rather, I have to delete one character to make the ‘create’ button available.

### Steps To Reproduce

1. Use input flyout to enter text into the name field without actually typing with the keyboard
2. See that the create file button is disabled

### Expected behavior

- Enable create file button no matter the input type

### Files Version

Version: 2.3.3.0 OS Version: 10.0.22000.795

### Windows Version

Windows 11 21H2 22000.795

### Relevant Assets/Logs

problem

### PR DESCRIPTION
**Pull request by hecksmosis**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #9563 

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

File: src/Files.App/Helpers/DynamicDialogFactory.cs [hunk 1]
```diff
@@ -103,6 +103,18 @@ public static DynamicDialog GetFor_RenameDialog()
 				}
 			};
 
+			inputText.TextChanged += (textBox, args) =>
+			{
+				if (!string.IsNullOrWhiteSpace(inputText.Text))
```
Comment (yaira2) [inline src/Files.App/Helpers/DynamicDialogFactory.cs:108]: Does this check for invalid characters?

Comment (hecksmosis) [reply src/Files.App/Helpers/DynamicDialogFactory.cs:108]: No, but as the text has already been changed we cannot revert to the old one.

Comment (hecksmosis) [reply src/Files.App/Helpers/DynamicDialogFactory.cs]: @yair100 What should we do then?

Comment (yaira2) [reply src/Files.App/Helpers/DynamicDialogFactory.cs:108]: > No, but as the text has already been changed we cannot revert to the old one.

The best thing might be to do a little refactoring and remove the TextChanging event so the behavior is consistent. We can disable the create button if the name isn't valid and display an explanation in a TextBlock.

File: src/Files.App/Helpers/DynamicDialogFactory.cs [hunk 2]
```diff
@@ -103,6 +103,18 @@ public static DynamicDialog GetFor_RenameDialog()
...
+				{
+					dialog.ViewModel.DynamicButtonsEnabled = DynamicDialogButtons.Primary | DynamicDialogButtons.Cancel;
+				}
+				else
+				{
+					dialog.ViewModel.DynamicButtonsEnabled = DynamicDialogButtons.Cancel;
+				}
```
Comment (QuaintMako) [inline src/Files.App/Helpers/DynamicDialogFactory.cs:115]: Can be simplified (here without the proper indentation):  
```suggestion
dialog.ViewModel.DynamicButtonsEnabled = string.IsNullOrWhiteSpace(inputText.Text) 
    ? DynamicDialogButtons.Cancel 
    : DynamicDialogButtons.Cancel | DynamicDialogButtons.Primary ;
```

File: src/Files.App/Helpers/DynamicDialogFactory.cs [hunk 3]
```diff
@@ -57,50 +57,26 @@ public static DynamicDialog GetFor_ShortcutNotFound(string targetPath)
...
-					else
-					{
-						dialog.ViewModel.DynamicButtonsEnabled = DynamicDialogButtons.Cancel;
-					}
-
-					tipText.Opacity = 0.0d;
-				}
+				tipText.Opacity = FilesystemHelpers.ContainsRestrictedCharacters(inputText.Text) ? 1.0d : 0.0d;
+				dialog!.ViewModel.DynamicButtonsEnabled = string.IsNullOrWhiteSpace(inputText.Text) || FilesystemHelpers.ContainsRestrictedCharacters(inputText.Text) 
+														? DynamicDialogButtons.Cancel : DynamicDialogButtons.Primary | DynamicDialogButtons.Cancel;
```
Comment (QuaintMako) [inline src/Files.App/Helpers/DynamicDialogFactory.cs:79]:
```suggestion
														? DynamicDialogButtons.Cancel 
														: DynamicDialogButtons.Primary | DynamicDialogButtons.Cancel;
```

Comment (QuaintMako) [reply src/Files.App/Helpers/DynamicDialogFactory.cs:79]: Jump a line.

File: src/Files.App/Helpers/DynamicDialogFactory.cs [hunk 4]
```diff
@@ -57,50 +57,27 @@ public static DynamicDialog GetFor_ShortcutNotFound(string targetPath)
...
-						dialog.ViewModel.DynamicButtonsEnabled = DynamicDialogButtons.Primary | DynamicDialogButtons.Cancel;
-					}
-					else
-					{
-						dialog.ViewModel.DynamicButtonsEnabled = DynamicDialogButtons.Cancel;
-					}
-
-					tipText.Opacity = 0.0d;
-				}
+				tipText.Opacity = FilesystemHelpers.ContainsRestrictedCharacters(inputText.Text) ? 1.0d : 0.0d;
```
Comment (yaira2) [inline src/Files.App/Helpers/DynamicDialogFactory.cs:77]: x:Load instead?

Comment (hecksmosis) [reply src/Files.App/Helpers/DynamicDialogFactory.cs:77]: It was used with opacity previously, that's why I decided to leave it that way.

Comment (hecksmosis) [reply src/Files.App/Helpers/DynamicDialogFactory.cs:77]: How would you use x:load here (sorry I am quite new to xaml)

Comment (yaira2) [reply src/Files.App/Helpers/DynamicDialogFactory.cs:77]: We can create a Boolean and bind the x:Load property of the text to the property.

Comment (hecksmosis) [reply src/Files.App/Helpers/DynamicDialogFactory.cs:77]: But there is no actual xaml for the text, is there?

Comment (hecksmosis) [reply src/Files.App/Helpers/DynamicDialogFactory.cs:77]: I think we can do this with Visibility if you prefer but as there is no xaml I don't think we can use x:Load

Comment (hecksmosis) [reply src/Files.App/Helpers/DynamicDialogFactory.cs:77]: Also, if we make the textblock unload or make it disappear, the dialog box size changes abruptly, and I don't think that looks good

Comment (yaira2) [reply src/Files.App/Helpers/DynamicDialogFactory.cs:77]: Forgot about the size changing, let's leave as is then.

**Review by yaira2 — DISMISSED** [review]
LGTM

**Comment by jiejasonliu** [comment]
this change will regress #7967, just wanted to point out if we're okay with that

**Comment by yaira2** [comment]
> this change will regress #7967, just wanted to point out if we're okay with that

Thanks for the heads up, this is okay.

File: src/Files.App/Helpers/DynamicDialogFactory.cs [hunk 5]
```diff
@@ -57,50 +57,27 @@ public static DynamicDialog GetFor_ShortcutNotFound(string targetPath)
...
+				dialog!.ViewModel.DynamicButtonsEnabled = string.IsNullOrWhiteSpace(inputText.Text) || FilesystemHelpers.ContainsRestrictedCharacters(inputText.Text)
```
Comment (QuaintMako) [inline src/Files.App/Helpers/DynamicDialogFactory.cs:78]: Shouldn't the test test also for `ContainsRestrictedFileName(name)` ?
If so, we could use here the `IsValidForFilename` method that is introduced in #10967 to simplify the test.

Comment (hecksmosis) [reply src/Files.App/Helpers/DynamicDialogFactory.cs:78]: I think we should merge #10967 first and then we can use the method

Comment (QuaintMako) [reply src/Files.App/Helpers/DynamicDialogFactory.cs]: @yair100

Comment (QuaintMako) [reply src/Files.App/Helpers/DynamicDialogFactory.cs:78]: @hecksmosis your move to update and use the method now!

Comment (hecksmosis) [reply src/Files.App/Helpers/DynamicDialogFactory.cs]: Done

File: src/Files.App/Helpers/DynamicDialogFactory.cs [hunk 6]
```diff
@@ -66,18 +66,18 @@ public static DynamicDialog GetFor_RenameDialog()
...
 			inputText.TextChanged += (textBox, args) =>
 			{
-				tipText.Opacity = FilesystemHelpers.ContainsRestrictedCharacters(inputText.Text) ? 1.0d : 0.0d;
-				dialog!.ViewModel.DynamicButtonsEnabled = string.IsNullOrWhiteSpace(inputText.Text) || FilesystemHelpers.ContainsRestrictedCharacters(inputText.Text) 
-														? DynamicDialogButtons.Cancel 
-														: DynamicDialogButtons.Primary | DynamicDialogButtons.Cancel;
+				tipText.Opacity = FilesystemHelpers.IsValidForFilename(inputText.Text) ? 0.0d : 1.0d;
+				dialog!.ViewModel.DynamicButtonsEnabled = FilesystemHelpers.IsValidForFilename(inputText.Text)
+														? DynamicDialogButtons.Primary | DynamicDialogButtons.Cancel
+														: DynamicDialogButtons.Cancel;
```
Comment (QuaintMako) [inline src/Files.App/Helpers/DynamicDialogFactory.cs:80]: We can win one test by stocking the value in a variable before hand and reusing them.
```suggestion
				var isInputValid = FilesystemHelpers.IsValidForFilename(inputText.Text);
				tipText.Opacity = isInputValid ? 0.0d : 1.0d;
				dialog!.ViewModel.DynamicButtonsEnabled = isInputValid 
														? DynamicDialogButtons.Primary | DynamicDialogButtons.Cancel
														: DynamicDialogButtons.Cancel;
```

**Review by QuaintMako — COMMENTED** [review]
That should be the last bit I can squeeze by being picky. LGTM otherwise.

**Review by yaira2 — APPROVED** [review]

**Pull request merged**